### ISSUE TITLE
mysql SET global and session not supported

### ISSUE THREAD

**Issue by willm** [sqlfluff/sqlfluff#6906]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The following is valid mysql  but raises a PRS error

```mysql
SET SESSION sql_mode='';
```

### Expected Behaviour

No parsing error is detected

### Observed Behaviour

== [test.sql] FAIL                                                                                                                    
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: "SET
                       | SESSION sql_mode='';"
WARNING: Parsing errors found and dialect is set to 'mysql'. Have you configured your dialect correctly?
All Finished 📜 🎉!

### How to reproduce

create a file `test.sql` with `SET SESSION sql_mode='';` and run `sqlfluff lint --dialect mysql test.sql`

### Dialect

mysql

### Version

3.4.0

### Configuration

I just have the defaults

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
